config/index.js

```diff
@@ -0,0 +1,57 @@
+const fs = require('fs')
+const path = require('path')
+const flat = require('flat')
+
+/**
+ * Load an environment config for the current environment, which is controlled by
+ * process.env.HASHI_ENV
+ */
+function loadHashiConfigForEnvironment() {
+  const env = process.env.HASHI_ENV || 'development'
+  const envConfigPath = path.join(process.cwd(), 'config', `${env}.json`)
+
+  return getHashiConfig(envConfigPath)
+}
+
+/**
+ * Load an environment config from a specific path.
+ */
+function getHashiConfig(configPath) {
+  try {
+    const baseConfigPath = path.join(process.cwd(), 'config', `base.json`)
+    const baseConfig = JSON.parse(fs.readFileSync(baseConfigPath))
+    const envConfig = JSON.parse(fs.readFileSync(configPath))
+
+    // Load the extended config, if no extends property is defined use the base config
+    let extendsConfig = baseConfig
+    if ('extends' in envConfig && envConfig.extends !== 'base') {
+      // Determine which config this environment is extending from
+      const extendsConfigPath = path.join(
+        process.cwd(),
+        'config',
+        `${envConfig.extends}.json`
+      )
+      extendsConfig = getHashiConfig(extendsConfigPath)
+    }
+
+    const extendsFlattened = flat(extendsConfig, { safe: true })
+
+    const envFlattened = flat(envConfig, {
+      safe: true,
+    })
+
+    // Because we are "flattening" the object, a simple spread should be sufficient here
+    const finalConfig = { ...extendsFlattened, ...envFlattened }
+
+    if (process.env.DEBUG_CONFIG) {
+      console.log(finalConfig)
+    }
+
+    return finalConfig
+  } catch (err) {
+    console.log('Error loading environment config:', err)
```

Comment:
I'm curious in what case we'd want to continue building the site if we were unable to load the config.

Reply:
Haha I had the same thought, I don't think we want to actually. I will adjust!